src/test/java/com/restapi/carMarket/controller/CarControllerUnitTest.java

```diff
@@ -0,0 +1,194 @@
+package com.restapi.carMarket.controller;
+
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.restapi.carMarket.api.CarController;
+import com.restapi.carMarket.exceptions.CarNotFoundException;
+import com.restapi.carMarket.exceptions.CarNotValidException;
+import com.restapi.carMarket.exceptions.RestExceptionHandler;
+import com.restapi.carMarket.model.Car;
+import com.restapi.carMarket.service.CarService;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.boot.test.json.JacksonTester;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+
+import static org.hamcrest.collection.IsCollectionWithSize.hasSize;
+import static org.hamcrest.core.Is.is;
+import static org.mockito.BDDMockito.given;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+import static org.mockito.Mockito.doThrow;
+import static org.springframework.http.MediaType.APPLICATION_JSON;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CarControllerUnitTest {
+    private MockMvc mockMvc;
+
+    @InjectMocks
+    private CarController carController;
+
+    @Mock
+    private CarService carService;
+
+    private JacksonTester<Car> jsonCar;
+
+    @Before
+    public void setup(){
+        JacksonTester.initFields(this, new ObjectMapper());
+
+        mockMvc = MockMvcBuilders.standaloneSetup(carController)
+                .setControllerAdvice(new RestExceptionHandler())
+                .build();
+    }
+
+    @Test
+    public void findCarsMustReturnAListOfCars() throws Exception {
+        List<Car> cars = new ArrayList<Car>();
+        cars.add(new Car("Porsche", "Panamera", 2017, 75000));
+        cars.add(new Car("BMW", "3 Series", 2019, 50000));
+
+        given(carService.getAllCarsOnMarket())
+                .willReturn(cars);
+
+        mockMvc.perform(get("/cars")
```

Comment:
This call is suitable for integration test when the whole server would start up and you start sending request to it. For a unit test it's expected to call the method, i.e.:
```
List<Cars> allCars = carController.findAll();
assertSame(cars, allCars);
```
In other words we would test that the controller delegates to the service and that's all.
Same is true for all other method tests in this class